wrench-service/src/main/java/com/izettle/wrench/service/WrenchService.kt

```diff
@@ -0,0 +1,90 @@
+package com.izettle.wrench.service
+
+import android.app.IntentService
+import android.content.ContentResolver
+import android.content.Intent
+import android.net.Uri
+import com.izettle.wrench.core.Bolt
+import com.izettle.wrench.core.WrenchProviderContract
+
+class WrenchService : IntentService("WrenchService") {
+
+    override fun onHandleIntent(intent: Intent?) {
+        if (intent != null) {
```

Comment:
Perhaps something like:
```
        intent?.let {
            val extras = it.extras
            for (key in extras.keySet()) {
                val value = extras.get(key)
                when (value) {
                    is Int -> updateInteger(contentResolver, key, value)
                    is String -> updateString(contentResolver, key, value)
                    is Boolean -> updateBoolean(contentResolver, key, value)
                }
            }
        }
```

looks better?

It uses a `intent?.let` to make sure it's not null
and also a `when` expression instead of lot of `if` statements.